models/surveys.go

```diff
@@ -729,3 +887,30 @@ func createStmt(sqlStatement string, db *sql.DB) (*sql.Stmt, error) {
 func (api *API) Close() {
 	api.AllSurveysStmt.Close()
 }
+
+// Roll back a given transaction and log any errors which occur
+func rollBack(tx *sql.Tx) {
+	err := tx.Rollback()
+	if err != nil {
+		logError("Error rolling back database transaction", err)
+	}
+}
+
+// Log and message and an error and send a 500 HTTP response
+func logErrorAndRespond500(w http.ResponseWriter, logMessage string, err error) {
+	logError(logMessage, err)
+	http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+}
+
+// Writes a 404 error response with a message
+func write404Response(w http.ResponseWriter, message string) {
+	response := NewRESTError("404", message)
```

Comment:
a bit of a nitpick but we should be using the http.statusnotfound constant here

Reply:
Discussed with Dave, this won't be done now because `NewRESTError` expects a `string` status code not an `int` type like the constant.